arch/cpu/nrf/sys/rtimer-arch.c

```diff
@@ -42,71 +43,74 @@
  * \file
  *         Implementation of the architecture dependent rtimer functions for the nRF
  * \author
+ *         Marcel Graber <marcel@clever.design>
  *         Yago Fontoura do Rosario <yago.rosario@hotmail.com.br>
  *
  */
 /*---------------------------------------------------------------------------*/
 #include "contiki.h"
+#include "sys/rtimer.h"
+#include "sys/soc-rtc.h"
 
 #include "nrf.h"
 #include "hal/nrf_timer.h"
 
-#ifdef NRF_RTIMER_CONF_TIMER_INSTANCE
-#define TIMER_INSTANCE NRF_RTIMER_CONF_TIMER_INSTANCE
-#else
-#define TIMER_INSTANCE 0
-#endif
-
-#if TIMER_INSTANCE == 0
-#define NRF_RTIMER_TIMER      NRF_TIMER0
-#define NRF_RTIMER_IRQn       TIMER0_IRQn
-#define NRF_RTIMER_IRQHandler TIMER0_IRQHandler
-#elif TIMER_INSTANCE == 1
-#define NRF_RTIMER_TIMER      NRF_TIMER1
-#define NRF_RTIMER_IRQn       TIMER1_IRQn
-#define NRF_RTIMER_IRQHandler TIMER1_IRQHandler
-#else
-#error Unsupported timer for rtimer
-#endif
-
+/*---------------------------------------------------------------------------*/
+static volatile rtimer_clock_t next_trigger = 0;
 /*---------------------------------------------------------------------------*/
 void
 rtimer_arch_init(void)
 {
-  nrf_timer_event_clear(NRF_RTIMER_TIMER, NRF_TIMER_EVENT_COMPARE0);
 
+  /* good place to initialize the RTC, used as base tick for rtimer */
+  soc_rtc_init();
+
+  nrf_timer_event_clear(NRF_RTIMER_TIMER, NRF_TIMER_EVENT_COMPARE0);
   nrf_timer_frequency_set(NRF_RTIMER_TIMER, NRF_TIMER_FREQ_62500Hz);
   nrf_timer_bit_width_set(NRF_RTIMER_TIMER, NRF_TIMER_BIT_WIDTH_32);
   nrf_timer_mode_set(NRF_RTIMER_TIMER, NRF_TIMER_MODE_TIMER);
   nrf_timer_int_enable(NRF_RTIMER_TIMER, NRF_TIMER_INT_COMPARE0_MASK);
-  NVIC_ClearPendingIRQ(NRF_RTIMER_IRQn);
-  NVIC_EnableIRQ(NRF_RTIMER_IRQn);
   nrf_timer_task_trigger(NRF_RTIMER_TIMER, NRF_TIMER_TASK_START);
 }
 /*---------------------------------------------------------------------------*/
+/**
+ *
+ * This function schedules a one-shot event with the AON RTC.
+ *
+ * This functions converts \e to a value suitable for the AON RTC.
```

Comment:
What does `\e` refer to?

Reply:
some garbage, I've used a lot ideas (and code) from the CC26x0 platform.